### PR TITLE
[In progress] Support TRT 7.1-8.0

### PR DESCRIPTION
**Pull request by oliviajain**
**Description**: Allow for any TRT version to be selected

**Motivation and Context**
- Validate perf test with multiple TRT Versions
- Combine mem_leak_test build with ort build

### CODE REVIEW

File: onnxruntime/python/tools/tensorrt/perf/build/Dockerfile.custom-trt-perf [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# --------------------------------------------------------------
```
Comment (oliviajain) [inline onnxruntime/python/tools/tensorrt/perf/build/Dockerfile.custom-trt-perf:1]: A few options. This is from https://github.com/nvidia/TensorRT. 
1. Checkout release from the Tensorrt repo + build ORT on top of it. ** can specify cuda version but doesn't have all combinations (how mem leak is using it right now) 
2. Add ubuntu dockerfiles to pipeline folder
3. Use manylinux (had trouble initially)

File: dockerfiles/Dockerfile.tensorrt [hunk 2]
```diff
@@ -25,6 +25,6 @@ RUN git clone --single-branch --branch ${ONNXRUNTIME_BRANCH} --recursive ${ONNXR
     cp onnxruntime/dockerfiles/LICENSE-IMAGE.txt /code/LICENSE-IMAGE.txt &&\
     cp onnxruntime/ThirdPartyNotices.txt /code/ThirdPartyNotices.txt &&\
     cd onnxruntime &&\
-    /bin/sh ./build.sh --parallel --cuda_home /usr/local/cuda --cudnn_home /usr/lib/x86_64-linux-gnu/ --use_tensorrt --tensorrt_home /workspace/tensorrt --config Release --build_wheel --update --build --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) &&\
+    /bin/sh ./build.sh --parallel --build_shared_lib --cuda_home /usr/local/cuda --cudnn_home /usr/lib/x86_64-linux-gnu/ --use_tensorrt --tensorrt_home /workspace/tensorrt --config Release --build_wheel --update --build --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) &&\
```
Comment (snnn) [inline dockerfiles/Dockerfile.tensorrt:28]: Why this change is needed?

Comment (oliviajain) [reply dockerfiles/Dockerfile.tensorrt:28]: For EP Perf Testing, we run in a docker container built from this dockerfile. The [memory leak testing](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/tensorrt/perf/mem_test) requires the onnxruntime shared library to exist. However, this is a work in progress and we may only run the memory leak test with a different dockerfile. I was previously trying to combine two separate build steps.

File: tools/ci_build/github/linux/docker/Dockerfile.ubuntu_cuda11_1_tensorrt7_2 [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# --------------------------------------------------------------
```
Comment (jywu-msft) [inline tools/ci_build/github/linux/docker/Dockerfile.ubuntu_cuda11_1_tensorrt7_2:1]: if we have cuda11_3, probably cuda11_1 isn't needed since 11.3 is compatible with all of 11.x

Comment (jywu-msft) [reply tools/ci_build/github/linux/docker/Dockerfile.ubuntu_cuda11_1_tensorrt7_2:1]: we may want to move to 11_4 though to align with everything else
